pkg/plugins/resources/k8s/native/api/v1alpha1/gateway_route_types_helpers.go

```diff
@@ -0,0 +1,63 @@
+//go:build gateway
+// +build gateway
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/pkg/model"
+	"github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/pkg/registry"
+)
+
+func (pt *GatewayRoute) GetObjectMeta() *metav1.ObjectMeta {
+	return &pt.ObjectMeta
+}
+
+func (pt *GatewayRoute) SetObjectMeta(m *metav1.ObjectMeta) {
+	pt.ObjectMeta = *m
+}
+
+func (pt *GatewayRoute) GetMesh() string {
+	return pt.Mesh
+}
+
+func (pt *GatewayRoute) SetMesh(mesh string) {
+	pt.Mesh = mesh
+}
+
+func (pt *GatewayRoute) GetSpec() map[string]interface{} {
+	return pt.Spec
+}
+
+func (pt *GatewayRoute) SetSpec(spec map[string]interface{}) {
+	pt.Spec = spec
+}
+
+func (pt *GatewayRoute) Scope() model.Scope {
+	return model.ScopeNamespace
+}
+
+func (l *GatewayRouteList) GetItems() []model.KubernetesObject {
+	result := make([]model.KubernetesObject, len(l.Items))
+	for i := range l.Items {
+		result[i] = &l.Items[i]
+	}
+	return result
+}
+
+func init() {
+	registry.RegisterObjectType(&mesh_proto.GatewayRoute{}, &GatewayRoute{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: GroupVersion.String(),
+			Kind:       "GatewayRoute",
+		},
+	})
+	registry.RegisterListType(&mesh_proto.GatewayRoute{}, &GatewayRouteList{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: GroupVersion.String(),
+			Kind:       "GatewayRouteList",
+		},
+	})
+}
```

Comment:
Suggest that we do the same as the gateway plugin - the code is always compiled (i.e. all these methods are always present), but the types are only registered when the build flag is set.